shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/screen/presentation/StudyPlanScreenReducer.kt

```diff
@@ -22,6 +23,31 @@ internal class StudyPlanScreenReducer(
             is StudyPlanScreenFeature.Message.Initialize -> {
                 handleInitializeMessage(state, message)
             }
+            is StudyPlanScreenFeature.Message.PullToRefresh -> {
+                val (widgetState, widgetActions)  = if (!state.studyPlanWidgetState.isRefreshing) {
+                    state.studyPlanWidgetState.copy(isRefreshing = true) to setOf(
+                        StudyPlanScreenFeature.InternalAction.StudyPlanWidgetAction(
+                            StudyPlanWidgetFeature.InternalAction.FetchStudyPlan(showLoadingIndicators = false)
+                        )
+                    )
+                } else {
```

Comment:
Why do you check only `studyPlanWidgetState.isRefreshing`, not all the embedded features state?